gnat2goto/driver/goto_utils.ads

```diff
@@ -59,6 +59,45 @@ package GOTO_Utils is
                   and then Kind (Index_Type) in Class_Type),
        Post => Kind (Compute_Memory_Op_Size'Result) = I_Op_Mul;
 
+   function Build_Function (Name : String; RType : Irep; Func_Params : Irep;
+                            FBody : Irep; A_Symbol_Table : in out Symbol_Table)
+                            return Symbol
+     with Pre => (Kind (RType) in Class_Type
+                  and then Kind (Func_Params) = I_Parameter_List
+                  and then Kind (FBody) = I_Code_Block);
```

Comment:
:pick: I think you can relax the precondition here to `Kind (FBody) in Class_Code`; Basically it doesn't need to be a code block specifically, it could also just be a single instruction (if `New_Function_Symbol_Entry` already has the other precondition, your call if you want to change that as well or just leave it as is, most functions have a `code_block` as value anyway so it's not a particularly big loss in flexibility in any case)

Reply:
My general approach was to keep the pre/post-condition as strict as possible until a sensible case emerged that violates them. But here, I guess you're right. Changed.